rfcs/text/0014_notification_service.md

```diff
@@ -0,0 +1,373 @@
+It's a bit complicated when it comes to sending notifications to a group of users.
+Sources work with different concepts of groups: Cloud - Teams, Kibana - Spaces. These concepts do not map to each other directly. 
+The notification service doesn't know about any user groups within the sources. So the group of users should unfold to a list of recipients before creating a notification. 
+Since the users in a group might have different roles and permissions, we have two options for security model implementation for group notifications:
+- A notification source, being a feature owner, enumerates a list of recipients before creating a notification.
+  It's technically impossible at the moment, but potentially this functionality might be added in [#80334](https://github.com/elastic/kibana/issues/80334).
+  With this model, it’s possible to create a notification that a user cannot act on when user permissions have been changed after notification creation.
+  It’s acceptable because it will be even more confusing if a notification disappears from the UI later due to changes in users permissions.
+  Having more use-cases, we can consider extending notification creation API to specify Kibana-specific permissions for the recipients
+  (send a notification to all the users with access to a feature X or space Y).
+  Although, it will require declaring Notification plugin dependency on other Kibana plugins, which can turn into circular dependency problems later.
+- Kibana calculates whether a user should receive a notification in runtime before showing one to the customer.
+  The only available option for the current Kibana Security model. Considering the future integration with UNS,
+  Kibana has to provide an HTTP API to check users' permissions are sufficient to see a notification.
+  Unified Notification center performs a call to HTTP API to filter out notifications before rendering them in Cloud UI. 
+  This option adds a significant runtime performance penalty. Moreover, it makes the rendering process dependent on Kibana instance availability. If one of them is temporarily unavailable, the UNS can't decide if a notification must be shown or not.
+
+### Notification System
+Abstracts the way notifications are created, stored, configured, and retrieved.
+It allows Kibana to adopt different strategies for on-premise and Cloud versions:
+- *LocalRepository* - a repository for on-premise version, it stores data within the current deployment. 
+- *RemoteRepository* - a repository for the Cloud version of Kibana. It stores data within the Unified Notification System deployed in the Cloud.
+In this case, a user can see in Kibana UI the external notifications created outside of Kibana.
+
+*Repository* interface:
+- `create(NotificationEvent)`- creates a notification in the Notification system. 
+- `get({ search_after?: number; size?: number; type_id?: string[] })` - retrieves last notifications for the current user.
```

Comment:
Does that return all notifications, or only the unread ones? Are there scenarios where we would need both?

Reply:
> Does that return all notifications,

all the notifications.

> Are there scenarios where we would need both?

You know, it might be the case for Notification Center UI. Probably we should extend API with `filter` parameter:
```js
filter: {
  type_id?: string[];
  group_id?: string[];
  read_status?: boolean;
}
```
WDYT?

Reply:
Yea, I think we will need that kind of filters quite soon. Your proposal seems fine